// @pulumi/mongodbatlas/cloudProviderAccess.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudProviderAccess = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Import
 *
 * The Cloud Provider Access resource can be imported using project ID and the provider name and mongodbatlas role id, in the format `project_id`-`provider_name`-`role_id`, e.g.
 *
 * ```sh
 *  $ pulumi import mongodbatlas:index/cloudProviderAccess:CloudProviderAccess my_role 1112222b3bf99403840e8934-AWS-5fc17d476f7a33224f5b224e
 * ```
 */
class CloudProviderAccess extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["atlasAssumedRoleExternalId"] = state ? state.atlasAssumedRoleExternalId : undefined;
            resourceInputs["atlasAwsAccountArn"] = state ? state.atlasAwsAccountArn : undefined;
            resourceInputs["authorizedDate"] = state ? state.authorizedDate : undefined;
            resourceInputs["createdDate"] = state ? state.createdDate : undefined;
            resourceInputs["featureUsages"] = state ? state.featureUsages : undefined;
            resourceInputs["iamAssumedRoleArn"] = state ? state.iamAssumedRoleArn : undefined;
            resourceInputs["projectId"] = state ? state.projectId : undefined;
            resourceInputs["providerName"] = state ? state.providerName : undefined;
            resourceInputs["roleId"] = state ? state.roleId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.projectId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectId'");
            }
            if ((!args || args.providerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerName'");
            }
            resourceInputs["iamAssumedRoleArn"] = args ? args.iamAssumedRoleArn : undefined;
            resourceInputs["projectId"] = args ? args.projectId : undefined;
            resourceInputs["providerName"] = args ? args.providerName : undefined;
            resourceInputs["atlasAssumedRoleExternalId"] = undefined /*out*/;
            resourceInputs["atlasAwsAccountArn"] = undefined /*out*/;
            resourceInputs["authorizedDate"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["featureUsages"] = undefined /*out*/;
            resourceInputs["roleId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CloudProviderAccess.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CloudProviderAccess resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CloudProviderAccess(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CloudProviderAccess.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudProviderAccess.__pulumiType;
    }
}
exports.CloudProviderAccess = CloudProviderAccess;
/** @internal */
CloudProviderAccess.__pulumiType = 'mongodbatlas:index/cloudProviderAccess:CloudProviderAccess';
